code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -1,13 +1,48 @@
+#define PURGING_REAGENTS list( \
+	/datum/reagent/medicine/c2/multiver, \
+	/datum/reagent/medicine/pen_acid, \
+	/datum/reagent/medicine/calomel, \
+	/datum/reagent/medicine/ammoniated_mercury, \
+	/datum/reagent/medicine/c2/syriniver, \
+	/datum/reagent/medicine/c2/musiver \
+)
+
 /datum/chemical_reaction/reagent_explosion
 	var/strengthdiv = 10
 	var/modifier = 0
 	reaction_flags = REACTION_INSTANT
 	reaction_tags = REACTION_TAG_EXPLOSIVE | REACTION_TAG_MODERATE | REACTION_TAG_DANGEROUS
 	required_temp = 0 //Prevent impromptu RPGs
-
-/datum/chemical_reaction/reagent_explosion/on_reaction(datum/reagents/holder, datum/equilibrium/reaction, created_volume)
-	default_explode(holder, created_volume, modifier, strengthdiv)
-
+	/// Only clear mob reagents in special cases
+	var/clear_mob_reagents = FALSE
+
+/datum/chemical_reaction/reagent_explosion/on_reaction(datum/reagents/holder, datum/equilibrium/reaction, created_volume, clear_mob_reagents)
+	/// If an explosive reaction clears mob reagents, it should always be a minimum power
+	if(ismob(holder.my_atom) && clear_mob_reagents)
+		if(round((created_volume / strengthdiv) + modifier, 1) < 1)
+			modifier += 1 - ((created_volume / strengthdiv) + modifier)
+	/// If this particular explosion doesn't automatically clear mob reagents as an inherent quality,
+	/// then we can still clear mob reagents with some mad science malpractice that shouldn't work but
+	/// does because omnizine is magic and also it's the future or whatever
+	if(ismob(holder.my_atom) && !clear_mob_reagents)
+		/// The explosion needs to be a minimum power to clear reagents: see above
+		var/purge_power = round((created_volume / strengthdiv) + modifier, 1)
+		if(purge_power >= 1)
+			var/has_purging_chemical = FALSE
+			/// They need one of the purge reagents in them
+			for(var/purging_chem as anything in PURGING_REAGENTS)
+				if(holder.has_reagent(purging_chem))
+					/// We have a purging chemical
+					has_purging_chemical = TRUE
+					break
+			/// Then we need omnizine! MAGIC!
+			var/has_omnizine = holder.has_reagent(/datum/reagent/medicine/omnizine)
+			if(has_purging_chemical && has_omnizine)
+				/// With all this medical "science" combined, we can clear mob reagents
```

Comment:
```suggestion
	// Only clear mob reagents in special cases
	var/clear_mob_reagents = FALSE

/datum/chemical_reaction/reagent_explosion/on_reaction(datum/reagents/holder, datum/equilibrium/reaction, created_volume, clear_mob_reagents)
	// If an explosive reaction clears mob reagents, it should always be a minimum power
	if(ismob(holder.my_atom) && clear_mob_reagents)
		if(round((created_volume / strengthdiv) + modifier, 1) < 1)
			modifier += 1 - ((created_volume / strengthdiv) + modifier)
	// If this particular explosion doesn't automatically clear mob reagents as an inherent quality,
	// then we can still clear mob reagents with some mad science malpractice that shouldn't work but
	// does because omnizine is magic and also it's the future or whatever
	if(ismob(holder.my_atom) && !clear_mob_reagents)
		// The explosion needs to be a minimum power to clear reagents: see above
		var/purge_power = round((created_volume / strengthdiv) + modifier, 1)
		if(purge_power >= 1)
			var/has_purging_chemical = FALSE
			// They need one of the purge reagents in them
			for(var/purging_chem as anything in PURGING_REAGENTS)
				if(holder.has_reagent(purging_chem))
					// We have a purging chemical
					has_purging_chemical = TRUE
					break
			// Then we need omnizine! MAGIC!
			var/has_omnizine = holder.has_reagent(/datum/reagent/medicine/omnizine)
			if(has_purging_chemical && has_omnizine)
				// With all this medical "science" combined, we can clear mob reagents
```